### ISSUE TITLE
inter-document xref links do not work after asciidoctor 1.5.6.2

### ISSUE THREAD

**Issue by aollier** [asciidoctor/asciidoctor-epub3#219]
Hello,

In the repositories [progit2](https://github.com/progit/progit2), [progit2-fr](https://github.com/progit/progit2-fr) and friends, we are using asciidoctor version 1.5.6.2 (previously 1.5.6.1) to build the tutorial for git (called **progit**) available on the [Git](https://git-scm.com) website, documentation section.

Starting from version 1.5.7, xref links do not work on any format: HTML, epub, mobi nor PDF.
This looks like to be a regression. We use links like &lt;chapter&gt;&num;&lt;section&gt;, which works perfectly until now on every format.

**Comment by mojavelinux** [comment]
FYI #220

**Comment by aollier** [comment]
Ok, thank you. I'll read very carefully, hoping that I'll find something that can help.

**Comment by mojavelinux** [comment]
After having a look at the progit2 project again, I think I see what the issue is. The progit book is being converted in compat mode because the doctitle uses the legacy two-line syntax. And in compat mode, interdoc xrefs are not available at all. That's because compat mode aims not to add any new syntax features from what was in AsciiDoc.py.

See https://github.com/asciidoctor/asciidoctor/blob/master/lib/asciidoctor/substitutors.rb#L750

**Comment by mojavelinux** [comment]
Yep, that's exactly the issue. No other change is necessary to make the interdoc xrefs work. However, not using compat mode does have other consequences. See https://asciidoctor.org/docs/migration/ But there's no reason anyone should be using compat mode at this point for documents that are still being modified.

**Comment by mojavelinux** [comment]
So, in the end, this is a support issue that pertains to core. So I'll move it back again now that I realize that.
